c7n/executor.py

```diff
@@ -40,17 +40,29 @@ def executor(name, **kw):
 
 
 class MainThreadExecutor(object):
+    """ For running tests.
+
+    async == True  -> catch exceptions and store them in the future.
+    async == False -> let exceptions bubble up.
+    """
+    
     # For Dev/Unit Testing with concurrent.futures
-    def __init__(self, *args, **kw):
+    def __init__(self, async=True, *args, **kw):
         self.args = args
         self.kw = kw
+        self.async = async
 
     def map(self, func, iterable):
         for args in iterable:
             yield func(args)
 
     def submit(self, func, *args, **kw):
-        return MainThreadFuture(func(*args, **kw))
+        try:
+            return MainThreadFuture(func(*args, **kw))
+        except Exception as e:
+            if self.async:
+                return MainThreadFuture(None, exception=e)
+            raise e
```

Comment:
this should be raise by itself without `e` otherwise the stack trace is altered from the original